index.js

```diff
@@ -542,11 +549,16 @@ SendStream.prototype.pipe = function pipe (res) {
   }
 
   // index file support
+  if (typeof this._index === 'function') {
```

Comment:
Can this check just be moved into the sendIndex method? It is a public method, and it's weird the event emit is in there but not this check.